File: backend/eTicketing.Model/Requests/LoginRequest.cs
using System.ComponentModel.DataAnnotations;

namespace eTicketing.Model.Requests;

public class LoginRequest
{
    [Required(ErrorMessage = "Email ili korisničko ime je obavezno")]
    public string EmailOrUsername { get; set; } = string.Empty;

    [Required(ErrorMessage = "Lozinka je obavezna")]
    public string Password { get; set; } = string.Empty;
}

File: frontend/desktop/lib/models/requests/login_request.dart
class LoginRequest {
  final String emailOrUsername;
  final String password;

  const LoginRequest({
    required this.emailOrUsername,
    required this.password,
  });

  Map<String, dynamic> toJson() => {
        'emailOrUsername': emailOrUsername,
        'password': password,
      };

  @override
  String toString() {
    return 'LoginRequest(emailOrUsername: $emailOrUsername, password: ****)';
  }
}
